### PR TITLE
Support mutable meta queries in a cluster

### LINKED ISSUE #5612: [0.10.0] DROP DATABASE does not work in a cluster
On a 3-node fully replicated cluster create a database on a node, and then make a write on that node. Finally, call `DROP DATABASE db`. 

Expected result:

 - The database information should be deleted from the meta store, and also all files wiped from each data node.

Actual result: 

 - The meta store is updated, but files are only removed on the the node where the `DROP DATABASE` command occurred.

### PR DESCRIPTION
**Pull request by e-dard**
Fixes #5612, #5573 and #5518. Partially #2755

Adds support in a multi-node cluster for the following commands:

 - `DROP DATABASE`;
 - `DROP MEASUREMENT`;
 - `DROP RETENTION POLICY`;
 - `DROP SERIES`.

The approach is as follows. Queries that involve modifying data on Data Nodes will be handed off by the `QueryExecuter` to the `MetaExecutor`. The `MetaExecutor` will then build appropriate messages and make RPC calls to all data nodes in the cluster. 

The node that received the query will apply any commands to its own `tsdb.Store` locally—they will not be sent over the network.

Errors encountered from remotely executed commands will be collected by the `MetaExecutor` and the first of these (if any) returned to the user.

Assuming no errors were returned by any data nodes, the node executing the original query will then make any calls necessary to the Meta Store.

Detailed Flow for `DROP DATABASE`:

- HTTP API receives query
- HTTP API passes query to cluster.QueryExecutor
- cluster.QueryExecutor confirms database exists
- cluster.QueryExecutor deletes database on local `tsdb.Store`
- cluster.QueryExecutor tells cluster.MetaExecutor to execute the statement
- cluster.MetaExecutor creates an executeStatementRequestMessage RPC command which is sent to cluster.Service on all **other** data nodes.
- cluster.Service dispatches command to cluster.Service.processExecuteStatementRequest
- cluster.Service.processExecuteStatementRequest unmarshals the request, parses the statement, and passes it to cluster.Service.executeStatement
- cluster.Service.executeStatement switches on the statement type and calls appropriate `tsdb.Store.DeleteDatabase`.

Errors propagate back from each node through RPC to the node that executed the statement on the cluster and that node returns failure if any node fails to execute.

- cluster.QueryExecutor tells meta.Client to delete the database

**TODO**
 - [x] Provide more useful errors?
 - [x] Fix some broken tests.
 - [x] Remove logging / println / commented out litter.
 - [x] Update CHANGELOG.
 - [x] Squash stuff down.

### CODE REVIEW

File: cluster/meta_executor.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+	if err != nil {
+		return err
+	}
+
+	conn, ok := c.(*pooledConn)
+	if !ok {
+		panic("wrong connection type in MetaExecutor")
+	}
+	// Return connection to pool by "closing" it.
+	defer func(conn net.Conn) { conn.Close() }(conn)
```
Comment (benbjohnson) [inline cluster/meta_executor.go]: Why is this wrapped in a function? Wouldn't this work the same:

```go
defer conn.Close()
```

File: cluster/meta_executor.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	for _, node := range nodes {
+		if m.Node.ID == node.ID {
+			continue // Don't execute statement on ourselves.
+		}
+
+		wg.Add(1)
+		go func(node meta.NodeInfo) {
+			defer wg.Done()
+			if err := m.executeOnNode(stmt, database, &node); err != nil {
+				errs <- err
```
Comment (benbjohnson) [inline cluster/meta_executor.go]: Can you add information about which node failed? I didn't see any info attached in `executeOnNode()`.

Comment (e-dard) [inline cluster/meta_executor.go]: good idea :+1:

**Comment by benbjohnson** [comment]
Minor comments, otherwise lgtm :+1:

File: cluster/meta_executor.go (hunk 1 above)
Comment (dgnorton) [inline cluster/meta_executor.go]: Somebody must have been smoking crack.  *whistles innocently*

File: cluster/meta_executor.go (hunk 2 above)
Comment (e-dard) [inline cluster/meta_executor.go]: @benbjohnson I've addressed this.

File: cluster/meta_executor.go [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+		DataNodes() ([]meta.NodeInfo, error)
+	}
+}
+
+// NewMetaExecutor returns a new initialized *MetaExecutor.
+func NewMetaExecutor() *MetaExecutor {
+	return &MetaExecutor{
+		timeout:        DefaultWriteTimeout,
+		pool:           newClientPool(),
+		maxConnections: 1000,
```
Comment (jwilder) [inline cluster/meta_executor.go]: Is this max connections per node?  If so, default is too high.  We use 3 per-node for writes.  Also, should be a constant like: https://github.com/influxdata/influxdb/blob/master/cluster/config.go#L21

File: cluster/meta_executor.go [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+	id  uint64
+	err error
+}
+
+func (e remoteNodeError) Error() string {
+	return fmt.Sprintf("partial success, node %d may be down (%s)", e.id, e.err)
+}
+
+// ExecuteStatement executes a single InfluxQL statement on all nodes in the cluster concurrently.
+func (m *MetaExecutor) ExecuteStatement(stmt influxql.Statement, database string) error {
```
Comment (jwilder) [inline cluster/meta_executor.go]: Is there any test coverage of this func?

**Comment by e-dard** [comment]
:trophy:

**Pull request merged**